x-pack/plugins/security_solution/server/endpoint/services/actions/clients/sentinelone/sentinel_one_actions_client.ts

```diff
@@ -249,28 +303,238 @@ export class SentinelOneActionsClient extends ResponseActionsClientImpl {
+              }
+            }
+            break;
+        }
+      }
+    }
+  }
+
+  /**
+   * Checks if the provided Isolate or Unisolate actions are complete and if so, then it builds the Response
+   * document for them and returns it. (NOTE: the response is NOT written to ES - only returned)
+   * @param actionRequests
+   * @param command
+   * @private
+   */
+  private async checkPendingIsolateOrReleaseActions(
+    actionRequests: Array<LogsEndpointAction<undefined, {}, SentinelOneIsolationRequestMeta>>,
+    command: ResponseActionsApiCommandNames & ('isolate' | 'unisolate')
+  ): Promise<LogsEndpointActionResponse[]> {
+    const completedResponses: LogsEndpointActionResponse[] = [];
+    const actionsByAgentId: {
+      [s1AgentId: string]: Array<
+        LogsEndpointAction<undefined, {}, SentinelOneIsolationRequestMeta>
+      >;
+    } = {};
+    const warnings: string[] = [];
+
+    // Create the `OR` clause that filters for each agent id and an updated date of greater than the date when
+    // the isolate request was created
+    const agentListQuery: QueryDslQueryContainer[] = actionRequests.reduce((acc, action) => {
+      const s1AgentId = action.meta?.agentId;
+
+      if (s1AgentId) {
+        if (!actionsByAgentId[s1AgentId]) {
+          actionsByAgentId[s1AgentId] = [];
+        }
+
+        actionsByAgentId[s1AgentId].push(action);
+
+        acc.push({
+          bool: {
+            filter: [
+              { term: { 'sentinel_one.activity.agent.id': s1AgentId } },
+              { range: { 'sentinel_one.activity.updated_at': { gt: action['@timestamp'] } } },
+            ],
+          },
+        });
+      } else {
+        // This is an edge case and should never happen. But just in case :-)
+        warnings.push(
+          `${command} response action ID [${action.EndpointActions.action_id}] missing SentinelOne agent ID, thus unable to check on it's status. Forcing it to complete as failure.`
+        );
+
+        completedResponses.push(
+          this.buildActionResponseEsDoc<{}, SentinelOneIsolationResponseMeta>({
+            actionId: action.EndpointActions.action_id,
+            agentId: Array.isArray(action.agent.id) ? action.agent.id[0] : action.agent.id,
+            data: { command },
+            error: {
+              message: `Unable to very if action completed. SentinelOne agent id ('meta.agentId') missing on action request document!`,
+            },
+          })
+        );
+      }
+
+      return acc;
+    }, [] as QueryDslQueryContainer[]);
+
+    if (agentListQuery.length > 0) {
+      const query: QueryDslQueryContainer = {
+        bool: {
+          must: [
+            {
+              terms: {
+                // Activity Types can be retrieved from S1 via API: `/web/api/v2.1/activities/types`
+                'sentinel_one.activity.type':
+                  command === 'isolate'
+                    ? [
+                        // {
+                        //    "id": 1001
```

Comment:
would you consider putting keeping these IDs in `enum` ?

Reply:
I'm not a big fan of `enum`s as they have some drawbacks. I did think about putting all of these codes in a `const` object, but at this time I am not seeing the need to use them outside of this code here, so I felt it was unnecessary.